src/main/drivers/bus_spi.c

```diff
@@ -124,170 +115,139 @@ uint32_t spiTimeoutUserCallback(SPI_TypeDef *instance)
+        uint32_t timeoutCheck = millis();
+        dmaTxBuffer[0] = reg;
+        dmaTxBuffer[1] = data;
+        dmaSpiTransmitReceive(dmaTxBuffer, dmaRxBuffer, 2, 1);
+        while(dmaSpiReadStatus != DMA_SPI_READ_DONE) {
+            if(millis() - timeoutCheck > GYRO_READ_TIMEOUT) {
+                //GYRO_READ_TIMEOUT ms max, read failed, cleanup spi and return 0
+                IOHi(bus->busdev_u.spi.csnPin);
+                return false;
             }
         }
-        else
-        {
-            IOLo(bus->busdev_u.spi.csnPin);
-            spiTransferByte(bus->busdev_u.spi.instance, reg);
-            spiTransferByte(bus->busdev_u.spi.instance, data);
-            IOHi(bus->busdev_u.spi.csnPin);
-        }
-    #else
+    } else {
         IOLo(bus->busdev_u.spi.csnPin);
         spiTransferByte(bus->busdev_u.spi.instance, reg);
         spiTransferByte(bus->busdev_u.spi.instance, data);
         IOHi(bus->busdev_u.spi.csnPin);
-    #endif
-
+    }
+#else
+    IOLo(bus->busdev_u.spi.csnPin);
+    spiTransferByte(bus->busdev_u.spi.instance, reg);
+    spiTransferByte(bus->busdev_u.spi.instance, data);
+    IOHi(bus->busdev_u.spi.csnPin);
+#endif
     return true;
 }
 
-FAST_CODE bool spiBusReadRegisterBuffer(const busDevice_t *bus, uint8_t reg, uint8_t *data, uint8_t length)
-{
-    #ifdef USE_DMA_SPI_DEVICE
-        if(USE_DMA_SPI_DEVICE == bus->busdev_u.spi.instance)
-        {
-            uint32_t timeoutCheck = millis();
-            dmaTxBuffer[0] = reg | 0x80;
-            dmaSpiTransmitReceive(dmaTxBuffer, dmaRxBuffer, length+1, 1);
-            while(dmaSpiReadStatus != DMA_SPI_READ_DONE)
-            {
-                if(millis() - timeoutCheck > GYRO_READ_TIMEOUT)
-                {
-                    //GYRO_READ_TIMEOUT ms max, read failed, cleanup spi and return 0
-                    IOHi(bus->busdev_u.spi.csnPin);
-                    return false;
-                }
+FAST_CODE bool spiBusReadRegisterBuffer(const busDevice_t *bus, uint8_t reg, uint8_t *data, uint8_t length) {
+#ifdef USE_DMA_SPI_DEVICE
+    if(USE_DMA_SPI_DEVICE == bus->busdev_u.spi.instance) {
+        uint32_t timeoutCheck = millis();
+        dmaTxBuffer[0] = reg | 0x80;
+        dmaSpiTransmitReceive(dmaTxBuffer, dmaRxBuffer, length + 1, 1);
+        while(dmaSpiReadStatus != DMA_SPI_READ_DONE) {
+            if(millis() - timeoutCheck > GYRO_READ_TIMEOUT) {
+                //GYRO_READ_TIMEOUT ms max, read failed, cleanup spi and return 0
+                IOHi(bus->busdev_u.spi.csnPin);
+                return false;
             }
-            memcpy(data, dmaRxBuffer+1, length);
-        }
-        else
-        {
-            IOLo(bus->busdev_u.spi.csnPin);
-            spiTransferByte(bus->busdev_u.spi.instance, reg | 0x80); // read transaction
-            spiTransfer(bus->busdev_u.spi.instance, NULL, data, length);
-            IOHi(bus->busdev_u.spi.csnPin);
         }
-    #else
+        memcpy(data, dmaRxBuffer + 1, length);
```

Comment:
Codacy found an issue: [Does not check for buffer overflows when copying to destination (CWE-120).  Make sure destination can always hold the source data. ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=6326343)